experimental/dds/tree2/src/feature-libraries/schemaSummarizer.ts

```diff
@@ -142,13 +145,15 @@ interface SchemaOp {
 export class SchemaEditor<TRepository extends StoredSchemaRepository>
 	implements StoredSchemaRepository
 {
-	private readonly codec: IJsonCodec<SchemaData, string>;
+	private readonly codec: IJsonCodec<SchemaData, Format>;
```

Comment:
An approach I've generally liked is to make external users of codecs type the encoded format as `unknown`. This works out without casting for the implementation within the codec, as we should be using type guards within the decode process (ex: the schema validation you have right now).

This avoids leaking the encoded format to files outside of persisted realm (though using it in tests is fine), and lets you avoid casts at usage sites of the codec e.g. on L238.

Reply:
Good points. I've tweaked `IJsonCodec` to allow loosening the input type of `decode` if desired (as it is here).